lib/spack/spack/operating_systems/windows_os.py

```diff
@@ -83,11 +85,46 @@ def compiler_search_paths(self):
                     os.path.join(str(os.getenv("ONEAPI_ROOT")), "compiler", "*", "windows", "bin")
                 )
             )
-        # Second strategy: Find MSVC via the registry
-        msft = winreg.WindowsRegistryView(
-            "SOFTWARE\\WOW6432Node\\Microsoft", winreg.HKEY.HKEY_LOCAL_MACHINE
-        )
-        vs_entries = msft.find_subkeys(r"VisualStudio_.*")
+
+        def try_query_registry(retry=False):
+            try:
+                # Second strategy: Find MSVC via the registry
```

Comment:
(minor) I think this specific line would be clearer if moved just above `def try_query_registry`

Reply:
Good point!